src/lambda/explorer/cloudFormationNodes.ts

```diff
@@ -86,6 +87,10 @@ export class CloudFormationStackNode extends AWSTreeNodeBase {
         return this.stackSummary.StackId
     }
 
+    public getArn(): string {
```

Comment:
Returning an empty string when an arn is expected is suprising. I would make the interface clearer here. If we can't guarantee that an Arn will always be available, we should either return `string | undefined` or throw an Error.

If an Error can be thrown, it should be documented on `AWSResourceNode#getArn` and caught in the command impl. If `undefined` can be returned, it should be handled in the command impl (and added to the signature).

Reply:
In practice, it shouldn't be possible for a resource to not have an ARN, but the types for those resources actually have the ARNs as optional fields. Given that, returning `string | undefined` makes sense here and should be handled in the command.

Related to [other Chris' comments](https://github.com/aws/aws-toolkit-vscode/pull/1010#discussion_r404193860) about error handling for the command.

Reply:
Yeah it depends on how much of a real possibility it is for these to be empty.

I would say if you've determined it shouldn't be possible but the types allow for it, check for undefined and throw an Error rather than returning undefined. It's a better way to convey to the caller that this is an exceptional event.

Reply:
Either way, I would add a comment on the interface describing the situation. That way, we can make informed decisions about how we want to handle the possibility of a missing arn for any future callers of this function. They may choose whether they want a fallback behavior based on the situation.

Reply:
Added comment on the interface and updated the implementations to throw if an ARN cannot be provided.